### DISCUSSION TITLE
highlight_linewidth_multipliers in DrawMoleculeWithHighlights

### DISCUSSION THREAD

**Discussion by keishk** [rdkit/rdkit#4972] [Q&A] [answered]
Hi

I am looking to change the bond highlight widths using:
class rdkit.Chem.Draw.rdMolDraw2D.MolDraw2D
DrawMoleculeWithHighlights

There is a param "highlight_linewidth_multipliers", and I have tried inputs like dictionary of bond_idx: val (like the highlight_radii param), however it has not worked. I was wondering whether this param can change the bond widths, and what the correct input format is.

Thanks

**Comment by DavidACosgrove** [comment]
Hi @keishk.  The highlight_linewidth_multipliers should take a dict of bond_idx: int, i.e. the multiplier is an integer.  What do you mean by "it has not worked"?  Can you please post a minimal example of the code and the results it produces.
Looking at it, it does feel like the multiplier would be more useful as a float.  I will discuss with @greglandrum to see if that would be a useful enhancement.

**Reply by keishk** [reply]
Hi @DavidACosgrove, thanks for your reply. There seems to be no effect of changing the param at the moment. For instance:

```
from collections import defaultdict
from rdkit import Chem
from rdkit.Chem.Draw import rdMolDraw2D
from IPython.display import SVG

chembl2314369=Chem.MolFromSmiles('N#CC(C#N)=C1C(=O)c2cccc3cccc1c23')
rings = chembl2314369.GetRingInfo()
colors = [(0.8,0.0,0.8),(0.8,0.8,0),(0,0.8,0.8),(0,0,0.8)]

athighlights = defaultdict(list)
arads = {}

for i, rng in enumerate(rings.AtomRings()):
    for aid in rng:
        athighlights[aid].append(colors[i])
        arads[aid] = 1
        
bndhighlights = defaultdict(list)
bnds = {}

for i, rng in enumerate(rings.BondRings()):
    for bid in rng:
        bndhighlights[bid].append(colors[i])
        
        # set bond multiple here
        bnds[bid]=500

d2d = rdMolDraw2D.MolDraw2DSVG(400,400)
d2d.DrawMoleculeWithHighlights(chembl2314369,'CHEMBL2314369',{},dict(bndhighlights),{},bnds)
d2d.FinishDrawing()
SVG(d2d.GetDrawingText())
```
Changing the int values in `bnds` has no effect it seems.

**Comment by DavidACosgrove** [comment, marked as answer]
This appears to be an issue with the Python layer.  The underlying C++ seems to be obeying the multiplier correctly.  Could you file it as a bug report, please?  I believe you can refer back to this discussion rather than copy and paste everything.  A multiplier of 20 should be enough to show an effect - 500 is very large, but that is not the cause of the problem, I don't think.
